components/account/home/AccountInformation.vue

```diff
@@ -103,69 +138,53 @@ async function update() {
   set(loading, false);
 }
 
-const waitTime = ref<number>(0);
-const loadingCheck = ref<boolean>(false);
-
-const { pause: pauseTimer, resume } = useIntervalFn(() => {
-  const wait = get(waitTime);
-  if (wait > 0) {
-    set(waitTime, wait - 1);
-  }
-  else {
-    pauseTimer();
-  }
-}, 1000, { immediate: false });
-
 async function handleCheckVATClick() {
   set(loadingCheck, true);
-  const result = await store.checkVAT();
-  await store.refreshVATCheckStatus();
+  const checkResult = await checkVAT();
+  await refreshVATCheckStatus();
 
-  if (typeof result === 'number') {
-    set(waitTime, result);
+  if ('seconds' in checkResult) {
+    const now = Math.round(Date.now() / 1000);
+    set(waitUntilTime, now + checkResult.seconds);
+    set(remainingTime, checkResult.seconds);
     pauseTimer();
-    if (result > 0) {
+    if (checkResult.seconds > 0) {
       resume();
     }
   }
-  else if (result) {
-    set(vatSuccessMessage, t('auth.signup.vat.verified'));
+  else if (checkResult.success) {
+    const status = get(account)?.vatIdStatus;
+    if (status === VatIdStatus.VALID) {
+      set(vatSuccessMessage, t('auth.signup.vat.verified'));
+      setTimeout(() => {
+        set(vatSuccessMessage, '');
+      }, 3000);
+    }
+  }
+  else {
+    set(vatErrorMessage, checkResult.message);
     setTimeout(() => {
-      set(vatSuccessMessage, '');
+      set(vatErrorMessage, '');
     }, 3000);
   }
   set(loadingCheck, false);
 }
 
-const [DefineVAT, ReuseVAT] = createReusableTemplate();
+function updateRemainingTime(): boolean {
+  const now = Math.round(Date.now() / 1000);
+  const endTime = get(waitUntilTime);
 
-const vatHint = computed(() => {
-  const wait = get(waitTime);
-  if (wait > 0) {
-    const formatted = formatSeconds(wait);
-    const time = `${(formatted.minutes || '00').toString().padStart(2, '0')}:${(formatted.seconds || '00').toString().padStart(2, '0')}`;
-    return t('auth.signup.vat.timer', { time });
+  if (endTime > now) {
+    set(remainingTime, endTime - now);
+    return true;
   }
+  return false;
+}
 
-  return t('auth.signup.customer_information.form.vat_id_hint');
-});
-
-const hideVATVerifyButton = computed(() => {
-  const status = get(account)?.vatIdStatus;
-  return status === VatIdStatus.NON_EU_ID;
-});
-
-const vatErrorMessage = computed(() => {
-  if (get(waitTime) > 0)
-    return '';
-  const status = get(account)?.vatIdStatus;
-  if (status === VatIdStatus.NOT_VALID) {
-    return t('auth.signup.vat.invalid');
-  }
-  else if (status === VatIdStatus.NOT_CHECKED) {
-    return t('auth.signup.vat.not_verified');
-  }
-  return '';
+onBeforeMount(() => {
```

Comment:
I think we should use `onMounted` instead.
I see a bug if the status is "Not Checked" and then we refresh the page on `http://localhost:3000/home/customer-information`. The button will be disabled.
But if you move to different tab and go back again, it's enabled now.